docs/fundamentals/executor/executor-api.md

```diff
@@ -226,55 +119,40 @@ class MyExecutor(Executor):
         print(kwargs)
 ```
 
-#### Multiple bindings
-
-To bind a method with multiple endpoints, you can use `@requests(on=['/foo', '/bar'])`. This allows
-either `f.post(on='/foo', ...)` or `f.post(on='/bar', ...)` to invoke that function.
-
 #### No binding
 
 A class with no `@requests` binding plays no part in the Flow. The request will simply pass through without any
 processing.
 
-(executor-method-signature)=
+### Method arguments
 
-### Method signature
+All Executor methods decorated by `@requests` need to follow the signature below for it to be used as a microservice inside a `Flow` (`async` is optional).
 
 Class method decorated by `@request` follows the signature below (`async` is optional):
 
 ```python
+from typing import Dict, Optional, Union, List
+from docarray import DocumentArray
+
 async def foo(docs: DocumentArray,
-        parameters: Dict,
-        docs_matrix: List[DocumentArray],
-        groundtruths: Optional[DocumentArray],
-        groundtruths_matrix: List[DocumentArray]) -> Optional[DocumentArray]:
+              parameters: Dict,
+              docs_matrix: List[DocumentArray]) -> Optional[Union[DocumentArray, Dict]]:
     pass
 ```
 
+Let's take a look at all these arguments:
 
-If the function is using `async` in its signature, it will be used as a coroutine and the regular `asyncio` functionality is available inside the function.
+- `docs`: A DocumentArray that is part of the request. As the nature of `Executor` is to wrap functionality related to `DocumentArray`, it is natural for
+it to be the main processing unit inside `Executor` methods. It is important to notice that these `docs` can be also changed in place as it would happen with 
+any other `list` object in a Python function
```

Comment:
```suggestion
- `docs`: A `DocumentArray` that is part of the request. Since the nature of `Executor` is to wrap functionality related to `DocumentArray`, it is usually the main processing unit inside `Executor` methods. It is important to notice that these `docs` can be also changed in place, just like it could happen with 
any other `list`-like object in a Python function.
```